File: slider/src/main/java/com/adsamcik/slider/ScaleFunctions/LinearScale.java
package com.adsamcik.slider.ScaleFunctions;

import com.adsamcik.slider.Sliders.Slider;

public class LinearScale {
	private LinearScale() {}

	public static Slider.IScale<Float> getFloatScale() {
		return (int progress, int maxProgress, Float min, Float max) -> min + (progress / (float)maxProgress) * (max - min);
	}

	public static Slider.IScale<Integer> getIntegerScale() {
		return (int progress, int maxProgress, Integer min, Integer max) -> progress + min;
	}
}

File: slider/src/main/java/com/adsamcik/slider/scaleFunctions/LinearScale.kt
package com.adsamcik.slider.scaleFunctions

import com.adsamcik.slider.Scale
import kotlin.math.roundToInt

internal object LinearScale {

	val floatScale: Scale<Float>
		get() = { position, min, max -> min + position * (max - min) }

	val integerScale: Scale<Int>
		get() = { position, min, max -> (position * (max - min)).roundToInt() + min }
}
